multinet/validation/__init__.py

```diff
@@ -33,3 +33,17 @@ class DuplicateKey(ValidationFailure):
     """Duplicate key detected when trying to create a table."""
 
     key: str
+
+
+class MetadataColumnKeyNotFound(ValidationFailure):
+    """A key in the column metadata was not found in the source data."""
+
+    key: str
+
+
+class IncompatibleMetadata(ValidationFailure):
```

Comment:
It's not that the metadata is incompatible, it's that the data is not convertible. I think a better name here will help future devs. Maybe something like `TypeConversionFailure`.

Reply:
1a9c194